### PR TITLE
added band gap (orbital) as training label and corresponding test case.

### PR DESCRIPTION
**Pull request by ouqi0711**
Merge the related code with band gap (orbital) label and corresponding test cases to bandgap branch.

### CODE REVIEW

File: deepks/scf/scf.py [hunk 1]
```diff
@@ -33,6 +33,25 @@ def t_make_pdm(dm, ovlp_shells):
...
+    """return projected density matrix by shell"""
+    # v stands for eigen values
+    pdm_shells = [dm.requires_grad_(True) for dm in t_make_pdm(dm, ovlp_shells)]
+    gvdm_shells = [t_batch_jacobian(t_shell_eig, dm, dm.shape[-1]) for dm in pdm_shells]
+    orbital_pdm_shells = [torch.einsum('rap,saq,ri,si->iapq', po, po, mo_coeff, mo_coeff) for po in ovlp_shells]
+    ips = [torch.einsum('iapq,avpq->iav', orbital_pdm, gvdm)
+                    for orbital_pdm, gvdm in zip(orbital_pdm_shells, gvdm_shells)]
+    return torch.cat(ips, dim=-1)
+
+def t_batch_jacobian(f, x, noutputs):
```
Comment (y1xiaoc) [inline deepks/scf/scf.py:46]: there is a function with same name at line 80. Is this a duplicated one?

File: deepks/scf/scf.py [hunk 2]
```diff
@@ -112,13 +131,20 @@ def energy_tot0(self, dm=None, h1e=None, vhf=None):
     def nuc_grad_method0(self):
         return super().nuc_grad_method()
 
+    def orbital0(self):
```
Comment (y1xiaoc) [inline deepks/scf/scf.py:134]: the naming convention is that `xxx0` correspond to the unmodified `xxx` method. Maybe you would like to rename it to `mo_energy0` and add `@property` decorator?

File: deepks/scf/scf.py [hunk 3]
```diff
@@ -112,13 +131,20 @@ def energy_tot0(self, dm=None, h1e=None, vhf=None):
...
+        return self.mo_energy - orbital_delta
+
     def get_veff(self, mol=None, dm=None, dm_last=0, vhf_last=0, hermi=1):
         """original mean field potential + correction potential"""
         if mol is None: 
             mol = self.mol
         if dm is None: 
             dm = self.make_rdm1()
-        tic = (time.process_time(), time.perf_counter())
+        tic = (time.clock(), time.perf_counter())
```
Comment (y1xiaoc) [inline deepks/scf/scf.py:147]: Please avoid using `clock()` method, which has been removed in python 3.8. The newest commit in the master branch fixed it.

File: deepks/scf/stats.py [hunk 4]
```diff
@@ -109,17 +123,31 @@ def load_stat(systems, dump_dir,
                 f_err.append(np.abs(lf - rf).mean((-1,-2)))
             except FileNotFoundError as e:
                 print("Warning! force file not found:", e, file=sys.stderr)
+        if with_o:
+            try:
+                ro = load_array(get_with_prefix(o_name, rbase, ".npy"))
+                lo = load_array(get_with_prefix("orbital", lbase, ".npy")).reshape(ro.shape)
+                [myc, myr] = ro.shape
+                ro1 = torch.tensor([myc,1])
```
Comment (y1xiaoc) [inline deepks/scf/stats.py:131]: can we avoid torch and use pure numpy? This file is intended to be pure numpy.

File: deepks/scf/stats.py [hunk 5]
```diff
@@ -2,6 +2,7 @@
 import sys
 import glob
 import numpy as np
+import torch
```
Comment (y1xiaoc) [inline deepks/scf/stats.py:5]: I intentionally avoid importing torch in this file to speed up the importing procedure. Can we avoid using torch?

**Review by y1xiaoc — CHANGES_REQUESTED** [review]
Thanks! It's a great PR! I only have several minor comments. Please see below.

File: deepks/scf/scf.py (hunk 1 above)
Comment (ouqi0711) [reply deepks/scf/scf.py:46]: Yes and sorry. The one at line 46 has now been removed.

File: deepks/scf/scf.py (hunk 3 above)
Comment (ouqi0711) [reply deepks/scf/scf.py:147]: Thanks and this has been fixed.

File: deepks/scf/stats.py (hunk 4 above)
Comment (ouqi0711) [reply deepks/scf/stats.py:131]: Yes, I've changed these to numpy operations.

File: deepks/scf/stats.py (hunk 5 above)
Comment (ouqi0711) [reply deepks/scf/stats.py:5]: Yes, line 5 has been removed.

File: deepks/scf/scf.py (hunk 2 above)
Comment (ouqi0711) [reply deepks/scf/scf.py:134]: Sure, I've changed orbital0 to mo_energy0 throughout the scripts and seems working fine.

**Comment by ouqi0711** [comment]
All fixes have been pushed to the master branch on ouqi0711/deepks-kit.
